include/stomp_moveit/trajectory_visualization.hpp

```diff
@@ -5,55 +5,113 @@
 #include <stomp_moveit/stomp_moveit_task.hpp>
 #include <stomp_moveit/conversion_functions.hpp>
 
+#include <std_msgs/msg/color_rgba.hpp>
+#include <tf2_eigen/tf2_eigen.hpp>
+#include <visualization_msgs/msg/marker.hpp>
+#include <visualization_msgs/msg/marker_array.hpp>
+
 namespace stomp_moveit
 {
 namespace visualization
 {
-PostIterationFn
-get_iteration_path_publisher(moveit_visual_tools::MoveItVisualTools& visual_tools,
-                             const moveit::core::JointModelGroup* group,
-                             const rviz_visual_tools::Colors color = rviz_visual_tools::TRANSLUCENT_LIGHT)
+
+namespace
 {
-  assert(group != nullptr);
+const rclcpp::Logger LOGGER = rclcpp::get_logger("stomp_moveit");
+const auto GREEN = [](const double& a) {
+  std_msgs::msg::ColorRGBA color;
+  color.r = 0.1;
+  color.g = 0.8;
+  color.b = 0.1;
+  color.a = a;
+  return color;
+};
+}  // namespace
+
+bool publishTrajectoryPoints(const robot_trajectory::RobotTrajectory& robot_trajectory,
```

Comment:
shouldn't we move this function into the anonymous namespace, also? I don't think that we want to expose (temporary?) visualization features with the public API.

Reply:
note about the signature: I would prefer to just build the marker and return it, so that the caller can call `publish()` on it. There is really no need at all for passing in the publisher.